### PR TITLE
Refactor Assignment & Questions Creation Form, Add Assignment/Question Editing Form, Integrate Assignment Service With Frontend

### PR DESCRIPTION
**Pull request by tryyang2001**
# Pull Request

## Description

This PR involves several changes:
1. Add a `description` field to `assignment` database schema so that tutors can provide assignment descriptions.
2. Change the UI of assignment creation form, and also added a question creation form (accessible when an assignment is newly created)
3. Consume `shadcn-ui` UI library with NextUI, so that we can now have FileUpload component. (I have created one under `/components/common` folder)
4. Update sidebar, add a "Create New Assignment" link in the menu, also created a dropdown for navigating to user page
5. Created a form for updating assignment detail and questions detail.
6. Added support for deleting individual question or deleting an assignment
7. Fix some logical bugs found in user context and user api-wrapper.
8. Fix some logical bugs found in AssignmentEditor and QuestionEditor.
9. Update assignment schema in the db to ensure questions are ordered by introducing `createdOn`.

## Related Issue(s)

#16 

## Screenshots (if applicable)

image
Create assignment form page.

image
Create questions page.

image
The new sidebar appearance

image
Assignment page now has a edit button and a delete button for tutor view (however, there is no access control for checking the user role now)

image
This is the appearance of edit form. 

## Checklist

- [x] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [x] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [x] I will take care of the merging and delete the side-branch after the PR is merged.

### CODE REVIEW

File: frontend/src/components/common/FileUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      setIsError(true);
+      onFileUpload("");
+      return;
+    }
+
+    if (!expectedFileTypes.includes(file.type)) {
+      setIsError(true);
+      onFileUpload("");
+      return;
+    }
```
Comment (junhonglow) [inline frontend/src/components/common/FileUpload.tsx:34]: When uploading a py file, file.type becomes a text/plain, so it becomes invalid.
image

**Review by junhonglow — COMMENTED** [review]
I wanted to try uploading a question, but I wasn't sure where to upload a sample question.
image

**Comment by tryyang2001** [comment]
> I wasn't sure where to upload a sample question

Sorry but what do you mean by this?

**Comment by junhonglow** [comment]
Bugs currently being fixed
1. Incorrect rendering of questions after performing delete operations
2. Order of questions shown in assignment overview is dependent on the order of last updated, rather than the order enforced by the creator.

**Review by junhonglow — APPROVED** [review]
LGTM

**Pull request merged**